lingcorpora/corpora/rus_poetic_corpus.py

```diff
@@ -0,0 +1,727 @@
+        :return: list of docs
+        """
+        sent = []
+        texts = []
+        query_list = []
+        ana_list = []
+        doc = {}
+        count = 0
+        for isStartComment in soup.find_all(text=self.isStartComment)[:int(n_docs)]:
+            if n_results == 0:
+                break
+            i += 1
+            meta = isStartComment.find_previous('span', class_='b-doc-expl').text.strip()
+            for text in isStartComment.find_all_next(text=True):
+
+                if self.isEndComment(text):
+                    isEnd = soup.find_all(text=self.isEndComment)[i]
+                    previous = isEnd.find_previous('span')
+                    while 'trim_up.html end' not in str(previous):
+                        if isinstance(previous, Tag):
+                            count -= 1
+                            if previous.attrs['class'][-1] == 'g-em':
+                                query_list.append([previous.text, count])
+                                if self.get_analysis:
+                                    ana_list.append(self.__ana(previous.attrs['explain']))
+                                if len(query_list) == len(self.query.split()):
+                                    n_results -= 1
+                                    query_list.reverse()
+                                    ana_list.reverse()
+                                    query = ' '.join(query_list[i][0] for i in range(len(query_list)))
+                                    doc['idxs'] = self.get_idxs(sent, query, query_list[0][1])
+                                    doc['meta'] = meta
+                                    doc['text'] = ' '.join(sent)
+                                    doc['analysis'] = ana_list
+                                    texts.append(doc)
+                                    doc = {}
+                                    ana_list = []
+                                    query_list = []
+                                if n_results == 0:
+                                    break
+                        if previous is None:
+                            break
+                        previous = previous.previous_sibling
+                    sent = []
+                    count = 0
+                    break
+
+                elif isinstance(text, Comment):
+                    continue
+
+                elif not text.strip():
+                    continue
+
+                elif self.__pattern.match(text):
+                    if sent:
+                        sent[-1] = sent[-1] + text.strip()
+                    else:
+                        sent.append(text)
+                else:
+                    sent.append(text.strip().replace(u'\xa0', u' '))
+
+        return texts
+
+    def __parse_text_page(self, link):
+        """
+        Parser for text pages when query is empty
+        """
+        req = requests.get(link)
+        time.sleep(randint(0, 3))
+        if req.status_code == 429:
```

Comment:
the same comments as above apply here. It could perhaps be more convenient to move the query logic into a separate function in order to get rid of the duplicate code